data/keystore/src/main/java/tech/pegasys/artemis/bls/keystore/KeyStoreLoader.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.bls.keystore;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidTypeIdException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.artemis.bls.keystore.model.KdfParam;
+import tech.pegasys.artemis.bls.keystore.model.KeyStoreData;
+
+/** Provide utility methods to load/store BLS KeyStore from json format */
+public class KeyStoreLoader {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final ObjectMapper OBJECT_MAPPER =
+      new ObjectMapper().registerModule(new KeyStoreBytesModule());
+
+  public static KeyStoreData loadFromFile(final Path keystoreFile)
+      throws KeyStoreValidationException {
+    checkNotNull(keystoreFile, "KeyStore path cannot be null");
+
+    try {
+      final KeyStoreData keyStoreData =
+          OBJECT_MAPPER.readValue(keystoreFile.toFile(), KeyStoreData.class);
+      keyStoreData.validate();
+      return keyStoreData;
+    } catch (final KeyStoreValidationException e) {
+      throw new KeyStoreValidationException("Error in parsing keystore: " + e.getMessage(), e);
+    } catch (final JsonParseException e) {
+      throw new KeyStoreValidationException("Error in parsing keystore: Invalid Json format", e);
+    } catch (final JsonMappingException e) {
+      throw convertToKeyStoreValidationException(e);
+    } catch (final IOException e) {
+      LOG.error("Error in parsing keystore: " + e.getMessage());
+      throw new KeyStoreValidationException("Error in parsing keystore: " + e.getMessage(), e);
+    }
+  }
+
+  private static KeyStoreValidationException convertToKeyStoreValidationException(
+      final JsonMappingException e) {
+    final String cause;
+    if (e.getCause() instanceof KeyStoreValidationException) {
+      cause = e.getCause().getMessage();
```

Comment:
Why are we getting `KeyStoreValidationException` wrapped in a `JsonMappingException`?  Shouldn't we just allow it to parse and then `validate` would fail?

Reply:
Updated and added code comment. This is wrapped because of deserializer raising our custom exception getting wrapped in jsonmappingexception.